lib/Logger.jsx

```diff
@@ -21,57 +21,49 @@ export default class Logger {
 
     /**
     * Ask the server to write the log message
-    *
-    * @param {String} message The message to write
-    * @param {Number} recentMessages A number of recent messages to append as context
-    * @param {Object|String} parameters The parameters to send along with the message
     */
-    logToServer(message, recentMessages, parameters = {}) {
-        // Optionally append recent log lines as context
-        let msg = message;
-        if (recentMessages > 0) {
-            msg += ' | Context:  ';
-            msg += JSON.stringify(this.get(recentMessages));
-        }
-
-        // Output the message to the console too.
-        if (this.isDebug) {
-            this.client(`${msg} - ${JSON.stringify(parameters)}`);
+    logToServer() {
+        // We do not want to call the server with an empty list or if all the lines has onlyFlushWithOthers=false
+        if (!_.any(this.logLines, l => !l.onlyFlushWithOthers)) {
+            return;
         }
 
         // We don't care about log setting web cookies so let's define it as false
-        const params = {parameters, message, api_setCookie: false};
-        this.serverLoggingCallback(params);
+        const promise = this.serverLoggingCallback({api_setCookie: false, logPacket: JSON.stringify(this.logLines)});
+        this.logLines = [];
+        if (!promise) {
```

Comment:
Why would this promise not exists?

Reply:
Because nowhere is checked that you need to pass it and I am not sure but I _think_ I saw a usage of it not passing it (but maybe I am making it up). Anyway, since this is a common lib I am trying to be extra cautious.

Reply:
Ok if we are being cautious then check to see if it's a promise before calling `.then()` on the return value?

Reply:
But also NAB I think this is ok.